internal/k8s/reconciler/gateway.go

```diff
@@ -149,6 +155,57 @@ func (g *K8sGateway) validateListenerConflicts() {
 	}
 }
 
+func (g *K8sGateway) validateGatewayIP(ctx context.Context) error {
+	service := g.serviceBuilder.Build()
+	if service == nil {
+		return nil
+	}
+
+	switch service.Spec.Type {
+	case corev1.ServiceTypeLoadBalancer, corev1.ServiceTypeClusterIP:
+		updated, err := g.client.GetService(ctx, types.NamespacedName{Namespace: service.Namespace, Name: service.Name})
+		if err != nil {
+			return err
+		}
+
+		if updated == nil {
+			g.status.Scheduled.NotReconciled = errors.New("service not found")
+			return nil
+		}
+
+		if service.Spec.Type == corev1.ServiceTypeLoadBalancer {
+			for _, ingress := range updated.Status.LoadBalancer.Ingress {
+				g.serviceReady = true
+				g.addresses = append(g.addresses, ingress.IP)
+			}
+		} else if service.Spec.Type == corev1.ServiceTypeClusterIP {
+			if updated.Spec.ClusterIP != "" {
+				g.serviceReady = true
+				g.addresses = append(g.addresses, updated.Spec.ClusterIP)
+			}
+		}
+	case corev1.ServiceTypeNodePort:
```

Comment:
For `serviceType: NodePort`, there isn't a great consistent way to guarantee access to the service from outside the k8s cluster. For now, we're putting the IP address of the nodes that the gateway pods are running on. The practitioner will have to understand that they might have to port forward into the cluster (in the case of Kind) or open firewall rules (in the case of GKE) in order to access the gateway from outside the cluster.

Reply:
Can we actually add this as a comment in this branch of the `switch` just so we know, in code, why we're using the `HostIP` status rather than (here a potentially empty) IP from the service itself?

Reply:
![](https://media.giphy.com/media/l41lZxzroU33typuU/giphy.gif)